game-core/src/main/java/games/strategy/triplea/delegate/IBattle.java

```diff
@@ -122,22 +122,22 @@ void unitsLostInPrecedingBattle(
   boolean isEmpty();
 
   /** Returns units which are dependent on the given units. */
-  Collection<Unit> getDependentUnits(Collection<Unit> units);
+  List<Unit> getDependentUnits(Collection<Unit> units);
```

Comment:
It's a list because that's what estimateStrengthDifference() takes, which is due to what the methods it calls take, which is due to what DiceRoll.getUnitPowerAndRollsForNormalBattles() takes and so on.

It's not obvious to me all of these need to be Lists and not Collections. I can take a look at changing them in a separate PR.

Reply:
I experimented with this before making the comment, switching it to `Collection` should work, you wind up making an API update to a half-dozen Pro methods and eventually once you get to the end it compiles.

Reply:
I sent https://github.com/triplea-game/triplea/pull/5592 for it.

We can land that one first and I can redo this on top of it, which then won't require the type changes here.

Reply:
@asvitkine  https://github.com/triplea-game/triplea/pull/5592 is merged